[cocoa] Change the opt-in cookie partitioning support to stable

sysrqb opened the pull request:
#### e0a1dc442c40fe4ec6bffd47bdcd4006c00cb374
<pre>
[cocoa] Change the opt-in cookie partitioning support to stable
<a href="https://bugs.webkit.org/show_bug.cgi?id=285343">https://bugs.webkit.org/show_bug.cgi?id=285343</a>
<a href="https://rdar.apple.com/142317056">rdar://142317056</a>

Reviewed by John Wilander.

Opt-in cookie partitioniong is currently only available on Cocoa platforms
where CFNetwork supports it.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformHave.h:

Canonical link: <a href="https://commits.webkit.org/288413@main">https://commits.webkit.org/288413@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1af19b3dc09c71a7c293cc7d96cd6bbd73bd0ad0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/83110 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/2751 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/37406 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/88219 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/34156 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/2828 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/10675 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/64713 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/22472 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/86165 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/2078 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/75576 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44993 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/1986 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/29776 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/33190 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/76095 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/73147 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/30475 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/89585 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/82155 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/10394 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/7509 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/73145 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/10623 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/71388 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/72369 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/16555 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/15291 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/1759 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12854 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/10347 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/15851 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/104564 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/10217 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/25308 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/13683 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/11986 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

johnwilander commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -5555,18 +5555,18 @@ OpportunisticSweepingAndGarbageCollectionEnabled:
   defaultValue:
     WebKitLegacy:
-      default: false
+      default: true
```
Do we have tests that make sure CHIPS works for Legacy? It's so rarely used that I don't think the benefit of enabling them there is large.

sysrqb replied:
I added a few tests that will be enabled on platforms where CHIPS is fully supported:

http/tests/storageAccess/grant-with-prompt-under-general-third-party-cookie-blocking-with-partitioned-cookies.https.html
http/tests/resourceLoadStatistics/only-partitioned-third-party-cookies.https.html
http/tests/resourceLoadStatistics/only-partitioned-cookies-after-redirect.https.html

johnwilander commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1883,12 +1883,3 @@
     || (PLATFORM(COCOA) && !PLATFORM(MAC))
 #define HAVE_X25519_ZERO_CHECKS 1
 #endif
-
```
We still have to disable on older macOS, right? For downlevels that is.

johnwilander replied:
But maybe it's gated on having some CFNetwork functionality so we're good already?

sysrqb replied:
Yes, in UnifiedWebPreferences we have a conditional:
```
condition: HAVE(ALLOW_ONLY_PARTITIONED_COOKIES)
```

so this preference is only available (and true) when that conditional evaluates to true.

johnwilander reviewed (approved):
Very happy to see this get enabled! Approved bar my comments.

Commit:
[cocoa] Change the opt-in cookie partitioning support to stable
https://bugs.webkit.org/show_bug.cgi?id=285343
rdar://142317056

Reviewed by John Wilander.

Opt-in cookie partitioniong is currently only available on Cocoa platforms
where CFNetwork supports it.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformHave.h:

Canonical link: https://commits.webkit.org/288413@main
